### PR TITLE
Refactor Filter

### PR DESCRIPTION
**Pull request by gusaiani**
Breakdown Filter into components:

· one for its Header, especially for mobile
· 4 for each param, currently Price, Area, Rooms, Neighborhoods

Reorganize filter state and make it Param-centric, so nest info around Price, Area etc, including param visibility

### CODE REVIEW

File: components/listings/index/filter/area.js [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+    }
+  }
+
+  render() {
+    const {
+      area,
+			handleMinAreaChange,
+			handleMaxAreaChange,
+			toggleVisibility,
+			handleClose
```
Comment (rhnonose) [inline components/listings/index/filter/area.js]: Essa formatação estranha é só no github?

File: components/listings/index/filter/neighborhoods.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  render() {
+    const {
+      neighborhoods,
+			options,
+      selectedOptions,
+			handleChange,
+			toggleVisibility,
+			handleClose
```
Comment (rhnonose) [inline components/listings/index/filter/neighborhoods.js]: Essa formatação estranha é só no github?

File: components/listings/index/filter/area.js (hunk 1 above)
Comment (gusaiani) [reply components/listings/index/filter/area.js]: @rhnonose só no Github ¯\_(ツ)_/¯

File: components/listings/index/filter/neighborhoods.js (hunk 2 above)
Comment (gusaiani) [reply components/listings/index/filter/neighborhoods.js]: yup

**Pull request merged**